File: static/typescript/0.1.0/src/dataset.ts
// Dataset: a container for tables.

import { Table } from './table';

export class Dataset {
  private tableByName: {[name: string]: Table} = {};

  constructor(
    readonly name: string,
    readonly version: string) { }

    addTable(table: Table) {
      if (table.name in this.tableByName) {
        throw Error(`Table named ${table.name} alreadu exists`);
      }
      this.tableByName[table.name] = table;
    }

    getTable(name: string): Table | undefined {
      return this.tableByName[name];
    }
}

File: static/python/0.1.0/src/dataset.py
class Dataset:
    def __init__(self, name: str, version: str):
        self.table_by_name = {}
        self.name = name
        self.version = version

    def add_table(self, table) -> None:
        if table.name in self.table_by_name:
            raise ValueError(f"Table named {table.name} already exists")
        self.table_by_name[table.name] = table

    def get_table(self, name: str):
        return self.table_by_name.get(name)
    